frontend/src/components/widgets/DataFrame/arrowUtils.test.ts

```diff
@@ -485,6 +488,94 @@ describe("getCellFromArrow", () => {
     })
   })
 
+  it("parses numeric timestamps into correct dates time values", () => {
+    const MOCK_TIME_COLUMN = {
+      ...TimeColumn({
+        id: "1",
+        name: "time_column",
+        title: "Time column",
+        indexNumber: 0,
+        isEditable: false,
+        isHidden: false,
+        isIndex: false,
+        isStretched: false,
+        arrowType: {
+          pandas_type: "time",
+          numpy_type: "object",
+        },
+      }),
+      getCell: jest.fn().mockReturnValue(getTextCell(false, false)),
+    }
+
+    // Create a mock arrowCell object with time data
+    const arrowCell = {
+      // Unix timestamp in microseconds Wed Sep 29 2021 21:13:20
+      // Our default unit is seconds, so it needs to be adjusted internally
+      content: BigInt(1632950000123000),
+      contentType: null,
+      field: {
+        type: {
+          unit: 2, // Microseconds
+        },
+      },
+      displayContent: null,
+      cssId: null,
+      cssClass: null,
+      type: "columns",
+    } as object as DataFrameCell
+
+    // Call the getCellFromArrow function
+    getCellFromArrow(MOCK_TIME_COLUMN, arrowCell)
+
+    // Check if the timestamp is adjusted properly
+    expect(MOCK_TIME_COLUMN.getCell).toHaveBeenCalledWith(
+      new Date("2021-09-29T21:13:20.123Z")
+    )
+  })
+
+  it("parses numeric timestamps into correct dates datetime values", () => {
```

Comment:
Should it be "into correct datetime values"?

Reply:
fixed it :)